pilot/pkg/config/kube/gateway/conditions.go

```diff
@@ -257,17 +257,25 @@ func reportListenerCondition(index int, l k8s.Listener, obj config.Config, condi
 			gs.Listeners = append(gs.Listeners, k8s.ListenerStatus{})
 		}
 		cond := gs.Listeners[index].Conditions
+		supported, valid := generateSupportedKinds(l)
+		if !valid {
+			conditions[string(k8s.ListenerConditionResolvedRefs)] = &condition{
+				reason:  string(k8s.ListenerReasonInvalidRouteKinds),
+				status:  metav1.ConditionFalse,
+				message: "Invalid route kinds",
```

Comment:
nit: could be helpful to make this more clear, like `Invalid route kinds: HTTPRoute cannot be used with protocol TLS` or something

Reply:
This could get a little wordy since in the general case because there could be several incompatible kinds that don't intersect. The spec says that the `supportedKinds` field must include the subset that are compatible, so comparing that to the requested kinds indicates which ones are invalid.